File: src/network/response/VaultInfoResponseBody.java
package org.elastos.hive.network.response;

import com.google.gson.annotations.SerializedName;

public class VaultInfoResponseBody extends HiveResponseBody {
    private String did;
    // TODO JAVA @SerializedName("max_storage")
    private long maxStorage;
    // TODO JAVA @SerializedName("file_use_storage")
    private long fileUseStorage;
    // TODO JAVA @SerializedName("db_use_storage")
    private long dbUseStorage;
    // TODO JAVA @SerializedName("modify_time")
    private long modifyTime;
    // TODO JAVA @SerializedName("start_time")
    private long startTime;
    // TODO JAVA @SerializedName("end_time")
    private long endTime;
    // TODO JAVA @SerializedName("pricing_using")
    private String pricingUsing;
    private String state;

    public String getDid() {
        return did;
    }

    public long getMaxStorage() {
        return maxStorage;
    }

    public long getFileUseStorage() {
        return fileUseStorage;
    }

    public long getDbUseStorage() {
        return dbUseStorage;
    }

    public long getModifyTime() {
        return modifyTime;
    }

    public String getModifyTimeStr() {
        return getDateStrByStamp(this.modifyTime);
    }

    public long getStartTime() {
        return startTime;
    }

    public String getStartTimeStr() {
        return getDateStrByStamp(this.startTime);
    }

    public long getEndTime() {
        return endTime;
    }

    public String getEndTimeStr() {
        return getDateStrByStamp(this.endTime);
    }

    public String getPricingUsing() {
        return pricingUsing;
    }

    public String getState() {
        return state;
    }

    public boolean isExisting() {
        return "running".equals(this.state);
    }
}

File: src/network/response/vaultinforesponsebody.ts
import { HiveResponseBody } from "./hiveresponsebody";

export class VaultInfoResponseBody extends HiveResponseBody {
    private did: string;
    // TODO JAVA @SerializedName("max_storage")
    private maxStorage: number;
    // TODO JAVA @SerializedName("file_use_storage")
    private fileUseStorage: number;
    // TODO JAVA @SerializedName("db_use_storage")
    private dbUseStorage: number;
    // TODO JAVA @SerializedName("modify_time")
    private modifyTime: number;
    // TODO JAVA @SerializedName("start_time")
    private startTime: number;
    // TODO JAVA @SerializedName("end_time")
    private endTime: number;
    // TODO JAVA @SerializedName("pricing_using")
    private pricingUsing: string;
    private state: string;

    public getDid(): string {
        return this.did;
    }

    public getMaxStorage(): number {
        return this.maxStorage;
    }

    public getFileUseStorage(): number {
        return this.fileUseStorage;
    }

    public getDbUseStorage(): number {
        return this.dbUseStorage;
    }

    public getModifyTime(): number {
        return this.modifyTime;
    }

    public getModifyTimeStr(): string {
        return this.getDateStrByStamp(this.modifyTime);
    }

    public getStartTime(): number {
        return this.startTime;
    }

    public getStartTimeStr(): string {
        return this.getDateStrByStamp(this.startTime);
    }

    public getEndTime(): number {
        return this.endTime;
    }

    public getEndTimeStr(): string {
        return this.getDateStrByStamp(this.endTime);
    }

    public getPricingUsing(): string {
        return this.pricingUsing;
    }

    public getState(): string {
        return this.state;
    }

    public isExisting(): boolean {
        return "running" === this.state;
    }
}
